src/kube-runtime/deploy/refresh.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+pushd $(dirname "$0") > /dev/null
+
+kubectl create configmap user-exit-spec-configuration --from-file=user-exit-spec.yaml --dry-run -o yaml | kubectl replace -f - || exit $?
```

Comment:
>create [](start = 8, length = 6)

The refresh is update? If so, use update operation instead of create

Reply:
Here we use kubectl replace -f - for the update refer to [update](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#updating-resources) . If we use in-place update operation such as patch, it will make the command complex. 

---
In reply to: [322045283](https://github.com/microsoft/pai/pull/3585#discussion_r322045283) [](ancestors = 322045283)

Reply:
How about [https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#kubectl-apply](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#kubectl-apply).

Replace will first delete then create, not so good.

---
In reply to: [322079743](https://github.com/microsoft/pai/pull/3585#discussion_r322079743) [](ancestors = 322079743,322045283)